langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -35,6 +38,12 @@
 public class Find {
 
     public static BArray find(BRegexpValue regExp, BString str, int startIndex) {
+        int strLength = str.length();
+        if (startIndex < 0 || strLength <= startIndex) {
```

Comment:
We'll have to handle number too large for the int input

Reply:
Add where applicable

Reply:
Had to update the method signatures to accept `long` instead of `int`. A similar approach to what's used in the `string:substring()` langlib is followed.

Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39238/commits/0701078442b74328565afaad98905ac0b7608d1d